### DISCUSSION TITLE
`#[must_use]`

### DISCUSSION THREAD

**Discussion by jRimbault** [rwf2/Rocket#2116] [Ideas] [answered]
https://github.com/SergioBenitez/Rocket/blob/7595450adc1aa3892004f02b606706597eb924e9/core/lib/src/error.rs#L61

Could this be replaced with a compile time [`#[must_use]`](https://doc.rust-lang.org/reference/attributes/diagnostics.html#the-must_use-attribute) ?

**Comment by SergioBenitez** [comment, marked as answer]
No. We need to know at runtime if the value has been inspected, but must_use is purely a (best effort) compile-time mechanism.
